### PR TITLE
Optimize UnifiedMap#detectOptional(Predicate2) to not use an iterator. Closes #442.

### LINKED ISSUE #442: Optimize  UnifiedMap#detectOptional(org.eclipse.collections.api.block.predicate.Predicate2) to not use an iterator.
See UnifiedMap#detect() for example implementation.

### PR DESCRIPTION
**Pull request by itohiro73**
Refer to issue #442.

### CODE REVIEW

File: eclipse-collections/src/main/java/org/eclipse/collections/impl/map/mutable/UnifiedMap.java [hunk 1]
```diff
@@ -1765,6 +1765,42 @@ else if (this.table[i] != null)
...
+                {
+                    return Optional.of(Tuples.pair(key, value));
+                }
+            }
+        }
+
+        return Optional.empty();
+    }
+
+     @Override
```
Comment (motlin) [inline eclipse-collections/src/main/java/org/eclipse/collections/impl/map/mutable/UnifiedMap.java]: There's extra whitespace on this last line. I figured the CheckStyle build would catch that?

File: unit-tests-java8/src/test/java/org/eclipse/collections/test/map/mutable/UnifiedMapNoIteratorTest.java [hunk 2]
```diff
@@ -94,14 +94,6 @@ public void RichIterable_getLast_empty_null()
...
     }
 
-    @Override
-    public void MapIterable_detectOptional()
-    {
-        /**
-         * TODO: {@link UnifiedMap#detectOptional(org.eclipse.collections.api.block.predicate.Predicate2)} should be optimized to not use an iterator
-         */
-    }
-
```
Comment (motlin) [inline unit-tests-java8/src/test/java/org/eclipse/collections/test/map/mutable/UnifiedMapNoIteratorTest.java]: Nice!

File: eclipse-collections/src/main/java/org/eclipse/collections/impl/map/mutable/UnifiedMap.java (hunk 1 above)
Comment (itohiro73) [reply eclipse-collections/src/main/java/org/eclipse/collections/impl/map/mutable/UnifiedMap.java]: Great catch indeed! 

Interesting why the build hasn't failed... 🤔

Comment (itohiro73) [reply eclipse-collections/src/main/java/org/eclipse/collections/impl/map/mutable/UnifiedMap.java]: Double checked checkstyle build locally, it doesn't fail either. 

Let me look into this separately why CheckStyle doesn't detect the extra white space...

I'll fix the white space itself soon.

**Comment by itohiro73** [comment]
Fixed the white space issue. Thanks!

**Comment by itohiro73** [comment]
Added RELEASE_NOTE_DRAFT.md, too.

**Comment by motlin** [comment]
LGTM, could you rebase though?

**Comment by itohiro73** [comment]
Thanks! Rebased.

**Pull request merged**